compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -1571,12 +1571,22 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             _ => traits::IsConstable::No,
         };
 
+        let is_repeatable = match element.kind {
+            hir::ExprKind::Call(func, _args) => match *self.node_ty(func.hir_id).kind() {
+                ty::FnDef(_, _) => true,
+                _ => false,
+            },
+            hir::ExprKind::Struct(_, _, _) => true,
+            _ => false,
+        };
```

Comment:
I'm not sure if we need this condition - checking for a function like you are here doesn't guarantee that it could be used with `from_fn`, as the function might only be able to be called once, and I don't know why a struct would make sense for this but not, for example, an enum. I'm not sure that there's necessarily a sensible condition to put here, and we might just want to always add the help.

Reply:
is it maybe too noisy to always add the help 😆
I rollback to the previous version.